### ISSUE TITLE
question about scrolling

### ISSUE THREAD

**Issue by JonathanMH** [hakimel/reveal.js#118]
Has anyone tried to create scrollable slides with this? I'd like to create something magazine-ish, that should work responsively, but I don't want to be limited by the screen size. Is there an option or a far better solution for this all?

I would like to use reveal.js for the great controls, transitions and mobile devices support. I'm trying to override some of the main.css and make it work through that. Is there an existing solution maybe?

**Comment by huashan** [comment]
Is it possible now in reveal.js to scroll down screen when a slide happen to be longer(or to say higher) than the screen height?

**Comment by raducoravu** [comment]
Same question here, could we scroll down through a slide if its content happens to have a height larger than the screen size?

**Comment by RafiKueng** [comment]
I guess as a hack it should be sufficient to the following to your `.slides` class?

```
overflow-y: auto;
overflow-x: hidden;
```

But this will add this behavior to all slides

**Comment by raducoravu** [comment]
Thanks for the suggestion, it seems that in order to make this work on my side I also had to specify a fixed height for the slide, otherwise the slide content was collapsed in a small vertical height with a scroll bar to the left. I'll keep looking into this when I have the time.
Ideally the library would add two up/down javascript buttons when the slide overflows the available vertical screen width.

**Comment by damontallen** [comment]
There is currently reveal.js is being integrated into IPython's nbviewer site and a question has come up about mobile behaviour with scrolling ([issue here](https://github.com/jupyter/nbviewer/issues/433) and [PR here](https://github.com/jupyter/nbviewer/pull/439#issuecomment-87916483)).  It is desirable to have both swipe action and the use of scrolling with large slides.  Is it possible to do both?  Perhaps scroll on slow gestures and move to the next slide on fast ones?

**Comment by RafiKueng** [comment]
(I don't really know what you are trying to achieve, I'm not dev of either project (but a user), nor mobile.. Just some ideas)

I thinks it's not the best idea to have two separate actions on the same gesture, only dependent on movement speed.. Maybe it's possible to have something like 1 finger swipe (resp. arrow-down key) vs 2 finger swipe (pgdn key), one does scoll, the other swipe? 

If it has to be the same gesture, then I'd say make it dependent on the duration of the initial press down, (press - hold - swipe/drag) and give some visual feedback maybe? really a ordinary swipe swipes, and a press down/ hold some time / then drag switches to scroll mode? (not sure whether it's technically possible, no experience with mobiles)

**Comment by damontallen** [comment]
Unfortunately my js experience is nearly non-existent as well.  I was just hoping for some clarification in response to issues brought up in the [PR](https://github.com/jupyter/nbviewer/pull/439#issuecomment-87916483).  Since I have a preference on the outcome but lack the expertise to provide a js solution I came here for advice.

**Comment by amitmutrezas** [comment]
I solved this with

.scrollable {
    overflow-y: auto  !important;
    overflow-x: hidden !important;
    height: 700px;
}

add the upper class to any any slide tag (section) to make it scrollable.

**Comment by G1305** [comment]
@raducoravu @JonathanMH @owenversteeg @hakimel @huashan @RafiKueng @amitmutrezas @damontallen @JonathanMH 

`<section class="scrollable">

and use the following css:

.scrollable {
    bottom: 0px;
    overflow-y: auto  !important;
    overflow-x: hidden !important;
}` THIS CODE WORKS FINE ONE SLIDE(page) ONLY.
but how can enable mouse scrolling feature as UP/DOWN arrow keys between sub-slides?
example:
`<div class="reveal">
			<div class="slides">
				<section>
					<section>
						<h3>SLIDE 1</h3>
					</section>
					<section>
						<h3>SLIDE 1.1</h3>
					</section>
					<section>
						<h3>SLIDE 1.2</h3>
					</section>
				
				<section>
					<section>
						<h3>SLIDE 2</h3>
					</section>
					<section>
						<h3>SLIDE 2.1</h3>
					</section>
					<section>
						<h3>SLIDE 2.2</h3>
					</section>`
I WANT scroll using mouse wheel between slide1 and slide 1.1 and slide 1.2
or next page - slide2 and slide 2.1 and slide 2.2

**Comment by cmhughes** [comment]
Thanks for this great tool :) 

For reference, I used

```css
.scrollable {
    overflow-y: auto  !important;
    overflow-x: hidden !important;
    height: 100%;
  }
```

When combined with `disableLayout: true,` this allows the slides to be zoomed to a high magnification without loss of content.

**Comment by axiomtutor** [comment]
I tried these solutions, although I applied them to all `section` tags rather than using a `scrollable` property.  I wanted to do it that way so that I didn't have to constantly fidget with the properties, based on how long the text was.   

However, now when it is applied to a smaller slide, I lose its vertical centering.  I.e. all the text starts flush at the top of the screen, even when there isn't a lot of text.  It sure looked a lot more attractive when a small amount of text had vertical centering.

Is there any way to have both?  That is to say, on small slides, keep vertical centering; but on large slides which exceed the vertical limits, have scrolling?
